src/heitech.zer0mqXt.core/ISocket.cs

```diff
@@ -4,24 +4,66 @@
 
 namespace heitech.zer0mqXt.core
 {
+    ///<summary>
+    /// Access to all underlying patterns of the zeroMQ Library
+    ///</summary>
     public interface ISocket : IDisposable
     { 
+        ///<summary>
+        /// Request a given TResult for the configuration and type of TRequest 
+        ///</summary>
         Task<TResult> RequestAsync<TRequest, TResult>(TRequest request)
             where TRequest : class, new()
             where TResult : class, new();
+
+        ///<summary>
+        /// Setup a nonblocking Responder to a Request of type of TRequest and return a TResult
+        ///</summary>
         void Respond<TRequest, TResult>(Func<TRequest, TResult> callback, CancellationToken cancellationToken = default)
             where TRequest : class, new()
             where TResult : class, new();
+
+        ///<summary>
+        /// Setup an asynchronous and nonblocking Responder for a Request of TRequest and return a TResult
+        ///</summary>
         void RespondAsync<TRequest, TResult>(Func<TRequest, Task<TResult>> callback, CancellationToken cancellationToken = default)
             where TRequest : class, new()
             where TResult : class, new();
 
+        ///<summary>
+        /// Publish a Message to all listening subscribers
+        ///</summary>
         Task PublishAsync<TMessage>(TMessage message)
             where TMessage : class, new();
 
+        ///<summary>
+        /// Register a non blocking Subscriber to a TMessage
```

Comment:
nit: in other comments we have `nonblocking`, but here and the next comment with a separated `non blocking`

Reply:
I´ll change it to non blocking for all of them